### PR TITLE
Ssh rust v6

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3445

Describe changes:
- Translates SSH C Parser to a Rust parser

This looks good

This is meant to be an exact translation. We want to add features later.
https://github.com/rusticata/ssh-parser might help for this

One difference though is the handling of end of line(s) at the end of the banner. This Rust version should be robust against packet splitting between CR and LF (which should be accepted by openssh), when the current version is vulnerable because end of lines skipping https://github.com/OISF/suricata/blob/master/src/app-layer-ssh.c#L235 is only done if the first packet contains the whole banner with end of lines. I made a suricata-verify test for this https://github.com/OISF/suricata-verify/pull/176

Modifies #4636 with style issues

Do we have to do something else than code for SSH app layer event rules Signature IDs ?

### CODE REVIEW

File: src/output-json-ssh.h [hunk 1]
```diff
@@ -28,6 +28,6 @@ void JsonSshLogRegister(void);
 
 #include "app-layer-ssh.h"
 
-void JsonSshLogJSON(json_t *js, SshState *tx);
+void JsonSshLogJSON(json_t *js, void *tx);
```
Comment (catenacyber) [inline src/output-json-ssh.h]: Should I not remove this function ? How come CI passed when it seems still used by alert-prelude.c ?

File: rules/ssh-events.rules [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# SSH app layer event rules
+#
+# SID's fall in the 2228000+ range. See https://redmine.openinfosecfoundation.org/projects/suricata/wiki/AppLayer
```
Comment (victorjulien) [inline rules/ssh-events.rules]: can you update this wiki page to match this?

Comment (catenacyber) [reply rules/ssh-events.rules]: Done

File: src/output-json-ssh.h (hunk 1 above)
Comment (jasonish) [reply src/output-json-ssh.h]: I don't think we enable prelude at compile time for any of the CI tests.  Probably should pick at least one build and turn it on.

Comment (catenacyber) [reply src/output-json-ssh.h]: And compiler/linker do not complain if we declare functions but do not define them...

**Pull request closed without merging**